diff -Nur linux-ldm-0.0.7.orig/Makefile linux-ldm-0.0.7/Makefile
--- linux-ldm-0.0.7.orig/Makefile	Sun Jun 23 20:41:22 2002
+++ linux-ldm-0.0.7/Makefile	Sun Jul  7 12:05:34 2002
@@ -9,8 +9,7 @@
 # General compile flags
 
 CFLAGS += -Wall
-CFLAGS += -g
-CFLAGS += -O2
+CFLAGS += $(OPT)
 CFLAGS += -D_FILE_OFFSET_BITS=64
 
 # LDM compile flags
@@ -23,11 +22,9 @@
 
 CFLAGS += -D__KERNEL__
 CFLAGS += -Wstrict-prototypes
-CFLAGS += -fomit-frame-pointer
 CFLAGS += -fno-strict-aliasing
 CFLAGS += -pipe
 CFLAGS += -mpreferred-stack-boundary=2
-CFLAGS += -march=$(shell uname -m)
 
 # ld flags
 
diff -Nur linux-ldm-0.0.7.orig/linux/fs/partitions/check.h linux-ldm-0.0.7/linux/fs/partitions/check.h
--- linux-ldm-0.0.7.orig/linux/fs/partitions/check.h	Thu Jan  1 01:00:00 1970
+++ linux-ldm-0.0.7/linux/fs/partitions/check.h	Sun Jul  7 12:05:34 2002
@@ -0,0 +1,16 @@
+/*
+ * add_partition adds a partitions details to the devices partition
+ * description.
+ */
+void add_gd_partition(struct gendisk *hd, int minor, int start, int size);
+
+typedef struct {struct page *v;} Sector;
+
+unsigned char *read_dev_sector(struct block_device *, unsigned long, Sector *);
+
+static inline void put_dev_sector(Sector p)
+{
+	page_cache_release(p.v);
+}
+
+extern int warn_no_part;
diff -Nur linux-ldm-0.0.7.orig/linux/fs/partitions/msdos.h linux-ldm-0.0.7/linux/fs/partitions/msdos.h
--- linux-ldm-0.0.7.orig/linux/fs/partitions/msdos.h	Thu Jan  1 01:00:00 1970
+++ linux-ldm-0.0.7/linux/fs/partitions/msdos.h	Sun Jul  7 12:05:34 2002
@@ -0,0 +1,9 @@
+/*
+ *  fs/partitions/msdos.h
+ */
+
+#define MSDOS_LABEL_MAGIC		0xAA55
+
+int msdos_partition(struct gendisk *hd, struct block_device *bdev,
+		    unsigned long first_sector, int first_part_minor);
+
diff -Nur linux-ldm-0.0.7.orig/test/Makefile linux-ldm-0.0.7/test/Makefile
--- linux-ldm-0.0.7.orig/test/Makefile	Sun Jun 23 20:41:22 2002
+++ linux-ldm-0.0.7/test/Makefile	Sun Jul  7 12:06:19 2002
@@ -11,6 +11,7 @@
 CFLAGS += -include extra.h
 CFLAGS += -I$(KERNEL)/include
 CFLAGS += -I$(KERNEL)/fs/partitions
+CFLAGS += -I../linux/fs/partitions
 
 all:	$(OUT)
 
--- linux-ldm-0.0.7/linux/fs/partitions/ldm.c.orig	Sun Jun 23 20:41:22 2002
+++ linux-ldm-0.0.7/linux/fs/partitions/ldm.c	Sun Jul  7 12:30:02 2002
@@ -34,6 +34,8 @@
 #include "check.h"
 #include "msdos.h"
 
+#undef memcpy
+
 #ifdef CONFIG_LDM_EXPORT_SYMBOLS
 #   define static
 #endif
--- linux-ldm-0.0.7/ldmutil/makefile.orig	Sun Jun 23 20:41:22 2002
+++ linux-ldm-0.0.7/ldmutil/makefile	Sun Jul  7 12:32:50 2002
@@ -2,7 +2,7 @@
 TARGET=ldmutil
 SRC=main.cpp ldm_db.cpp diskio.cpp ldm_parse.cpp ptypenames.cpp
 OBJ=$(SRC:%.cpp=%.o)
-FLAGS=-O3
+FLAGS=$(OPT)
 LDFLAGS=-Xlinker --strip-all
 LIB=
 INC=
